app/controllers/report_templates_controller.rb

```diff
@@ -8,16 +8,30 @@ def documentation_anchor
 
   def generate
     find_resource
-    headers["Cache-Control"] = "no-cache"
-    headers["Content-Disposition"] = %(attachment; filename="#{@template.suggested_report_name}")
-    safe_render(@template)
+    @composer = ReportComposer.from_ui_params(params)
+  end
+
+  def schedule_report
+    find_resource
```

Comment:
it can't be done, if I set `before_action :find_resource` here, it overrides the actions specified in parent controller (templates controller), I'd have to copy the list here which is worse as it would get out of sync quickly, since you're second who asked, I'll add a comment above this for future